### PR TITLE
feat(leanpkg/main.lean): allow passing extra arguments to Lean

### PR DESCRIPTION
**Pull request by dselsam**
Note: there is some extra complexity because I wanted to distinguish `leanpkg-args` and `lean-args`.
If anyone who works on `leanpkg` would prefer not to, I can change it.

### CODE REVIEW

File: leanpkg/leanpkg/main.lean [hunk 1]
```diff
@@ -45,22 +45,21 @@ assg ← solve_deps d,
...
 exec_cmd {
   cmd := "lean",
   args := (match manifest.timeout with some t := ["-T", repr t] | none := [] end) ++
-    ["--make"] ++ manifest.effective_path,
+    ["--make"] ++ manifest.effective_path ++ lean_args,
   env := [("LEAN_PATH", none)]
 }
 
-def build := configure >> make
+def build (lean_args :list string) := configure >> make lean_args
```
Comment (gebner) [inline leanpkg/leanpkg/main.lean]: The colon in the type ascription should be followed by a space.

File: leanpkg/leanpkg/main.lean [hunk 2]
```diff
@@ -149,32 +148,32 @@ configure
 def usage := "
 Usage: leanpkg <command>
 
-configure       download dependencies
-build           download dependencies and build *.olean files
-test            download dependencies and run test files
+configure            download dependencies
+build -- <lean-args> download dependencies and build *.olean files
```
Comment (gebner) [inline leanpkg/leanpkg/main.lean]: I think we should put brackets (the `[]` variety) around `-- <lean-args>` to indicate that the extra arguments are optional.

File: leanpkg/leanpkg/main.lean (hunk 1 above)
Comment (dselsam) [reply leanpkg/leanpkg/main.lean]: Thanks. Sorry I missed the space -- the `S-SPC` key-binding in Emacs eats some of my spaces (I change it to TAB on my branches).

File: leanpkg/leanpkg/main.lean (hunk 2 above)
Comment (dselsam) [reply leanpkg/leanpkg/main.lean]: Agreed.

**Comment by gebner** [comment]
Now `leanpkg` fails without printing the usage message when called with no arguments:
```
$ leanpkg
split_cmdline_args: no args provided
```

**Comment by dselsam** [comment]
> Now `leanpkg` fails without printing the usage message when called with no arguments:

Amended.

**Pull request merged**